### DISCUSSION TITLE
bug: args_0 undefined

### DISCUSSION THREAD

**Discussion by Brandon-Perry** [trpc/trpc#3445] [Q&A]
### Provide environment information

System:
    OS: macOS 12.4
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Memory: 36.45 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.10.0 - ~/Desktop/predictionBook/backend/node_modules/.bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.19.2 - /usr/local/bin/npm
    Watchman: 2022.10.31.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 108.0.5359.124
    Firefox: 108.0.1
    Safari: 15.5

### Describe the bug

Client package
```
"@craco/craco": "^6.4.5",
    
    "@trpc/client": "^9.27.4",
    "@trpc/server": "^9.27.4",
    "@types/jest": "^27.0.1",
    "@types/node": "^16.7.13",
    "typescript": "^4.4.2",
    "zod": "^3.19.1"
```

Backend package 
```
 "@trpc/client": "^9.27.4",
    "@trpc/server": "^9.27.4",
    "dotenv": "^16.0.3",
    "express": "^4.18.2",
    "node": "^18.10.0",
    "nodemon": "^2.0.20",
    "jest": "^29.3.1",
    "zod": "^3.19.1"
```

The issue is that in my apiTests.test.ts file (using Jest), my client.mutation call is not working as intended. 
For some reason its not adopting the types from the router, despite context being provided. In the example below, it says that "Argument of type '{ email?: string; password?: string; }' is not assignable to parameter of type 'undefined'"
The part that's tripping me is that this works just fine in my front-end code, where this exact same call is made and works.

const launchClient = createTRPCClient<LaunchRouter>({
    url: `http://localhost:${PORT}/v1/launch`,
});

test("Login and Signup", async()=> {

    const email = randEmail()
    const password = randPassword() + '@'
    const name = randUserName()

    const apiReq:api.LoginRequest = {email, password}
    await expect(await launchClient.mutation('login',apiReq)).rejects.toThrow("FORBIDDEN")
    
})

Edit:

Here are my tsconfigs
client

{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": ["src"]
}

backend

{
  "compilerOptions": {
    "target": "es2015" 
    "module": "CommonJS" 
    "removeComments": true /* Do not emit comments to output. */,
    "noEmit": true /* Do not emit outputs. */,
    "moduleResolution": "node" /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */,
    "esModuleInterop": true 
    "experimentalDecorators": true /* Enables experimental support for ES7 decorators. */,
    "skipLibCheck": true /* Skip type checking of declaration files. */,
    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */
  },
  "include":["src","tests"]
}

### Link to reproduction

https://stackblitz.com/github/trpc/examples-next-minimal-starter?file=README.md

### To reproduce

I could not reproduce the bug

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by KATT** [comment]
Paste your tsconfig(s) please

**Reply by Brandon-Perry** [reply]
@KATT sorry it took so long to reply, ts config is added

**Reply by KATT** [reply]
I doubt it's a bug on our end but we don't have the bandwidth to confirm that.

Since this is a v9 issue we won't spend time on chasing this. Update to v10 and let us know if you have issues there.
